DashAI/back/api/api_v1/endpoints/jobs.py

```diff
@@ -97,35 +95,11 @@ async def enqueue_runner_job(params: JobParams, db: Session = Depends(get_db)):
     dict
         dict with the new job on the database
     """
-    try:
-        run: Run = db.get(Run, params.run_id)
-        if not run:
-            raise HTTPException(
-                status_code=status.HTTP_404_NOT_FOUND, detail="Run not found"
-            )
-    except exc.SQLAlchemyError as e:
-        log.exception(e)
-        raise HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            detail="Internal database error",
-        ) from e
-    job = Job(
-        func=execute_run,
-        type=JobType.runner,
-        kwargs={"run_id": params.run_id, "db": db},
-    )
+    params.db = db
+    job: BaseJob = component_registry[params.job_type]["class"](**params.model_dump())
+    job.set_status_as_delivered()
```

Comment:
This code must be surrounded by a try catch block to throw an HTTPException in case the method fails.